### ISSUE TITLE
[BUG] 400 InvalidParameter on VMSS list instances

### ISSUE THREAD

**Issue by weidongxu-microsoft** [Azure/azure-sdk-for-java#27991]
**Describe the bug**

We observed occasional errors from AzureSDK 2.12.0, and we would really appreciate it if you could help us to get more insights.
```
com.azure.resourcemanager.compute.models.ApiErrorException: Status code 400, "{
"error": {
"code": "InvalidParameter",
"message": "The request URL is not valid.",
"target": "$skiptoken"
}
}": The request URL is not valid.
```
Relevant stacktrace is attached.
 
The exception was thrown when we were calling [this API](https://docs.microsoft.com/en-us/rest/api/compute/virtual-machine-scale-set-vms/list) using Azure Java SDK:
```
vmss
      .virtualMachines()
      .list()
      .stream()
      .collect(Collectors.toList())
```

***Exception or Stack Trace***
Add the exception log and stack trace if available

**To Reproduce**

***Code Snippet***
Add the code snippet that causes the issue.

**Expected behavior**

**Screenshots**

**Setup (please complete the following information):**
 - IDE: [e.g. IntelliJ]
 - Library/Libraries: [e.g. com.azure:azure-core:1.16.0 (groupId:artifactId:version)] azure-resourcemanager-compute:2.12.0
 - Java version: [e.g. 8] 
 - App Server/Environment: [e.g. Tomcat, WildFly, Azure Function, Apache Spark, Databricks, IDE plugin or anything special]
 - Frameworks: [e.g. Spring Boot, Micronaut, Quarkus, etc] 

If you suspect a dependency version mismatch (e.g. you see `NoClassDefFoundError`, `NoSuchMethodError` or similar), please check out [Troubleshoot dependency version conflict article](https://aka.ms/azsdk/java/dependency/troubleshoot) first. If it doesn't provide solution for the problem, please provide:
- verbose dependency tree (`mvn dependency:tree -Dverbose`)
- exception message, full stack trace, and any available logs

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by weidongxu-microsoft** [comment]
First call is good
```
GET https://management.azure.com/subscriptions/ec0aa5f7-9e78-40c9-85cd-535c6305b380/resourceGroups/rg-weidxu/providers/Microsoft.Compute/virtualMachineScaleSets/vmss1/virtualMachines?api-version=2021-11-01
```

Wait for quota increasement in test subscription.

**Comment by weidongxu-microsoft** [comment]
still waiting for quota...

**Comment by weidongxu-microsoft** [comment]
Finally
```
WESTUS3, BS Series / 500
```

@XiaofeiCao

**Comment by weidongxu-microsoft** [comment]
azure-core 1.31.0 should have the fix shipped.

**Comment by XiaofeiCao** [comment]
Exceeds virtual machine limits of 100... 
`WESTUS3, Standard_B1s, 250`:
```
{
  "error" : {
    "code" : "OperationNotAllowed",
    "message" : "Unable to create or update Virtual Machine Scale Set 'vmss92626' as it exceeds the total limit of '100' Virtual Machine instances."
  }
}
```

**Comment by XiaofeiCao** [comment]
>[reactor-http-nio-2] INFO com.azure.resourcemanager.compute.implementation.VirtualMachineScaleSetVMsClientImpl$VirtualMachineScaleSetVMsService.listNext - <-- 200

Seems they've fixed it.

**Comment by weidongxu-microsoft** [comment]
Fixed in azure-core 1.31.0

SDK 2.18.0 should have the above azure-core as dependency.

Detail in https://github.com/Azure/azure-sdk-for-java/issues/29248
